components/calendar/index.js

```diff
@@ -64,60 +58,70 @@ const Calendar = React.createClass({
 
   buildEvents: function (events) {
     let id = 0;
-    let lastEpochDays = 0;
     let daysEvents = [];
+    let previousEventDate;
 
     let preDays = [];
     let postDays = [];
     let preDaysSection = '';
 
     events.forEach(function (event, index) {
```

Comment:
I'll definitely look into this while dealing with the other issues, and failing that I'll do it independent of issues later.